tools/edit_llm.py

```
@@ -0,0 +1,752 @@
+                raise ToolError("view_range must be [start, end]")
+            start, end = vrange
+            if start < 1 or end < -1:
+                raise ToolError("Invalid view_range values")
+            end = len(lines) if end == -1 else end
+        snippet = "\n".join(self._numbered(lines[start - 1 : end], offset=start))
+        return ToolResult(output=snippet)
+
+    def _numbered(self, lines: List[str], *, offset: int = 1) -> List[str]:
+        return [f"{idx + offset:6}\t{line}" for idx, line in enumerate(lines)]
+
+    def _snippet(self, text: str, start: int, end: int) -> str:
```

Ruff (0.12.2)

Unused method argument: `end`

(ARG002)